src/Linker.php

```diff
@@ -0,0 +1,227 @@
+<?php
+
+namespace PhpSchool\WorkshopManager;
+
+use Composer\IO\IOInterface;
+use League\Flysystem\Filesystem;
+use PhpSchool\WorkshopManager\Entity\Workshop;
+use PhpSchool\WorkshopManager\Exception\WorkshopNotInstalledException;
+
+/**
+ * Class Linker
+ * @author Michael Woodward <mikeymike.mw@gmail.com>
+ */
+final class Linker
+{
+    /**
+     * @var ManagerState
+     */
+    private $state;
+
+    /**
+     * @var Filesystem
+     */
+    private $filesystem;
+
+    /**
+     * @var bool
+     */
+    private $useSystem;
```

Comment:
Ok cool makes sense, maybe add some text somewhere describing the process of checking the dirs :)